spec/src/main/asciidoc/architecture.adoc

```diff
@@ -324,12 +326,7 @@ There should be a single `MetricRegistry` instance shared between all applicatio
 of different registries while responding to metric export requests. It is up to the application server whether it will allow sharing
 of metrics between different applications (for example, if there's a reusable metric in one application, another might want to reuse it).
 
-==== Implementation notes:
-Constructors of the `MetricID` class from the API code already handle adding the `_app` tag automatically
-when they detect that there is a property named `mp.metrics.appName` available from the `org.eclipse.microprofile.config.Config` instance
-available in the current context. If no such property exists or if the value is empty, no tag will be appended.
-
-Generally, the responsibility of the application server implementation will be to append a property `mp.metrics.appName` to the
-`org.eclipse.microprofile.config.Config` instance of each application during deployment time, its value being the web context root of the application
-or the JAR name. This can be achieved for example by adding a custom `ConfigSource` with an ordinal less than 100, because
-the `ConfigSource` that reads properties `META-INF/microprofile-config.properties` has an ordinal of 100, and this needs to have higher priority.
+It should also be noted that because MP Config supports different configuration property values per each application, it can potentially
+happen that each application will see a different value of `mp.metrics.tags` which denotes the global tags. In such case, the
+implementation must make sure that the value of `mp.metrics.tags` is read only on the global level and thus metrics from all applications
+will get the same global tags.
```

Comment:
Do we need to restrict people configuring mp.metrics.tags to only do so via env vars or system properties in multi-app servers? With a single-app server it could be just as valid to use an mp.config.properties file, but in a multi-app server it seems simpler to just mandate that env vars or system properties be used to configure mp.metrics.tags.

Reply:
I would agree. But the current wording kinda does that, doesn't it? It says that `mp.metrics.tags` will only be resolved on the global level, so it makes no sense to define it per application. I wouldn't go into too much more detail in here, as this is beyond the scope of MicroProfile.

Reply:
I think it's taken me a few reads through to get what you meant, but I agree you've said basically what I said in a different way.